### PR TITLE
Add lib for mysql server and mysql database

### PR DESCRIPTION
**Pull request by mikaelkrief**
### Description

Add azurerm test librairies for:
- mysql server + mysql servers
- mysql server database + mysql server databases

include:
- librairies code
- terraform code for create mysql server and database
- associate tests
- documentation
- update the Readme with doc links

### CODE REVIEW

**Comment by dmccown** [comment]
Thank you for your contribution! I'll work with @r-fennell and make sure one of us gets this reviewed.

File: docs/resources/azurerm_mysql_databases.md.erb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+Filters the results to include only those databases that match the given
+name. This is a string value.
+
+    describe azurerm_mysql_databases(resource_group: 'rg', server_name: 'server').where{ name.eql?('production-database') } do
+      it { should exist }
+    end
+    
+* `location`
+
+### location
```
Comment (r-fennell) [inline docs/resources/azurerm_mysql_databases.md.erb]: Is `location` a filterable field based on the response? I don't see location listed at https://docs.microsoft.com/en-us/rest/api/mysql/databases/listbyserver#databaselistresult

File: docs/resources/azurerm_mysql_server.md.erb [hunk 2]
```diff
@@ -0,0 +1,121 @@
+
```
Comment (r-fennell) [inline docs/resources/azurerm_mysql_server.md.erb]: This doc appears to have suffered a small formatting issue! It seems to be missing its header section e.g.
```

---
title: About the azurerm_mysql_server Resource
platform: azure
---

.... etc
```

File: docs/resources/azurerm_mysql_servers.md.erb [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+---
+
+# azurerm\_mysql\_servers
+
+Use the `azurerm_mysql_servers` InSpec audit resource to test properties and configuration of Azure MySQL Servers.
+<br />
+
+## Azure REST API version
+
+This resource interacts with version `2018-06-01-preview` of the Azure Management API. For more
```
Comment (r-fennell) [inline docs/resources/azurerm_mysql_servers.md.erb]: `2017-12-01`

File: lib/environment_file.rb [hunk 4]
```diff
@@ -4,6 +4,7 @@ class EnvironmentFile
     network_watcher
     msi
     sql
+    mysql
```
Comment (r-fennell) [inline lib/environment_file.rb]: This is fine to bundle with the `sql` option. You can update your controls to `only_if { ENV['SQL'] }`

File: terraform/azure.tf [hunk 5]
```diff
@@ -634,3 +634,49 @@ resource "azurerm_sql_database" "sql-database" {
...
+  upper   = false
+}
+
+resource "azurerm_mysql_server" "mysql" {
+  name                = "${random_string.mysql_server.result}-mysqlsvr"
+  location            = "westeurope"
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+
+  sku {
+    name     = "${var.sku_name}"
```
Comment (r-fennell) [inline terraform/azure.tf]: If the variables being referenced here (in `variables.tf`) are not reused, we should simply hardcode the values in `azure.tf` until they're needed in multiple places. I dont expect we will need to reuse any the MySQL values.

File: terraform/variables.tf [hunk 6]
```diff
@@ -73,3 +73,71 @@ variable "public_vm_count" {
 variable "sql-server-version" {
   default = "12.0"
 }
+
+
+# Azure mysql
```
Comment (r-fennell) [inline terraform/variables.tf]: As above, it's fine to hardcode most/all of these straight into the terraform resource, unless they are being used in several places!

File: test/integration/verify/controls/azurerm_mysql_database.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
+resource_group = attribute('resource_group', default: nil)
+mysql_server_name = attribute('mysql_server_name', default: nil)
+mysql_db_name = attribute('mysql_database_name', default: nil)
+
+control 'azurerm_mysql_database' do
+  only_if { ENV['MYSQL'] }
+
+  describe azurerm_mysql_database(resource_group: resource_group,
+                                server_name: mysql_server_name,
+                                database_name: mysql_db_name) do
```
Comment (r-fennell) [inline test/integration/verify/controls/azurerm_mysql_database.rb]: Tab alignment :)

**Comment by mikaelkrief** [comment]
@r-fennell  Thanks, I fixed and pushed all this feedbacks

**Comment by r-fennell** [comment]
Hi,
I'm closing this PR as a duplicate of https://github.com/inspec/inspec-azure/pull/161 which builds on this work.
Thanks @mikaelkrief !

**Pull request closed without merging**